pkg/kubelet/kuberuntime/kuberuntime_manager_test.go

```diff
@@ -603,3 +603,43 @@ func TestSyncPodWithInitContainers(t *testing.T) {
 		t.Errorf("expected %q, got %q", expectedContainers, actual)
 	}
 }
+
+func TestPruneInitContainers(t *testing.T) {
+	fakeRuntime, _, m, err := createTestRuntimeManager()
+	assert.NoError(t, err)
+
+	status := &kubecontainer.PodStatus{
+		ContainerStatuses: []*kubecontainer.ContainerStatus{
+			{Name: "init2", ID: kubecontainer.ContainerID{ID: "init2_1"}, State: kubecontainer.ContainerStateExited},
+			{Name: "init2", ID: kubecontainer.ContainerID{ID: "init2_0"}, State: kubecontainer.ContainerStateExited},
+			{Name: "init1", ID: kubecontainer.ContainerID{ID: "init1_2"}, State: kubecontainer.ContainerStateExited},
+			{Name: "init1", ID: kubecontainer.ContainerID{ID: "init1_1"}, State: kubecontainer.ContainerStateExited},
+			{Name: "init1", ID: kubecontainer.ContainerID{ID: "init1_0"}, State: kubecontainer.ContainerStateExited},
+			{Name: "init3", ID: kubecontainer.ContainerID{ID: "init3_0"}, State: kubecontainer.ContainerStateRunning},
+		},
+	}
+
+	var fakeContainers []apitest.FakeContainer
+	for _, s := range status.ContainerStatuses {
+		fakeContainers = append(fakeContainers, apitest.FakeContainer{
+			*s,
```

Comment:
I don't think this can compile.
The status in `FakeContainer` is `runtimeApi.ContainerStatus`.
The status your are using is `kubecontainer.ContainerStatus`.

In fact the original `PruneInitContainerTest` is wrong. See https://github.com/kubernetes/kubernetes/pull/35018/commits/7cd8487ca4175ea093c018fde60642a5035a3a84.